### DISCUSSION TITLE
My Teslamate has no location information, and sometimes there is no energy consumption and energy efficiency data during the trip.

### DISCUSSION THREAD

**Discussion by BRUCELEEGYX** [teslamate-org/teslamate#3598] [General]
When it was first deployed, there was location information in the itinerary. Later, for some unknown reason, the location information never appeared again. The location information in the itinerary was always empty. In addition, there are often no calculations of energy consumption and energy efficiency information during the trip. Do you have any friends with the same problem?
![1](https://github.com/teslamate-org/teslamate/assets/61728672/b0a2b459-7584-4183-af8f-15286a0a1c1c)

**Comment by cwanja** [comment]
Locations - I recall OpenStreetMaps is blocked in China. Believe you have to use a proxy to get out and get the data. 

Efficiency - notice the snowflake. And the short drives do not derive efficiency on short drives in cold weather.

**Reply by BRUCELEEGYX** [reply]
Yes, the map shows proxy, I know this. But the problem now is that the location data of the record is empty. Energy efficiency can look at the picture I cut. There is a long trip, about 20 or 30 minutes, but there is still no energy efficiency data.

**Reply by dyxyl** [reply]
The efficiency data is only shown when this is true:

```
duration_min > 1 AND distance > 1 AND ( 
      start_position.usable_battery_level IS NULL OR end_position.usable_battery_level IS NULL OR
      (end_position.battery_level - end_position.usable_battery_level) = 0 
    )
```

so the most likely problem is that `end_position.battery_level` is different from `end_position.usable_battery_level` probably due to a cold battery.
